src/pages/index.js

```diff
@@ -13,53 +15,181 @@ import {
+const changeProfileAvatarPopup = new PopupWithForm(
+  "#change-profile-avatar-modal",
+  handleChangeProfileAvatarFormSubmit
+);
+const editProfilePopup = new PopupWithForm(
+  "#edit-modal",
+  handleEditProfileFormSubmit
+);
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+const deleteCardPopup = new PopupWithConfirmation("#delete-card-modal");
 
-const addCardPopup = new PopupWithForm("#add-card-modal", (value) => {
-  const cardData = {
-    name: value.Title,
-    link: value.URL,
-  };
-  renderCard(cardData);
-  addCardPopup.close();
-});
-addCardPopup.setEventListeners();
+// handle functions
+function handleCardDelete(cardId) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setLoading(true, "Deleting...");
+    api
+      .deleteCard(cardId._id)
+      .then(() => {
+        deleteCardPopup.close();
+        cardId.removeCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        deleteCardPopup.setLoading(false, "Yes");
+      });
+  });
+}
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+function handleChangeProfileAvatarFormSubmit(url) {
+  changeProfileAvatarPopup.setLoading(true, "saving...");
+  api
+    .updateAvatar(url)
+    .then((data) => {
+      userInfo.setAvatar(data.avatar);
+      changeProfileAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => changeProfileAvatarPopup.setLoading(false, "Save"));
+}
 
-const editProfilePopup = new PopupWithForm("#edit-modal", (profileData) => {
-  userInfo.setUserInfo(profileData.name, profileData.description);
-  editProfilePopup.close();
-});
-editProfilePopup.setEventListeners();
+function handleEditProfileFormSubmit(data) {
+  editProfilePopup.setLoading(true, "Saving...");
+  api
+    .profileUpdate(data)
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        about: userData.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => editProfilePopup.setLoading(false, "Save"));
+}
+
+function handleAddCardFormSubmit(card) {
+  addCardPopup.setLoading(true, "Saving...");
+  api
+    .addCard(card)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => addCardPopup.setLoading(false, "Create"));
+}
 
+function handleCardLike(cardId) {
+  if (!cardId.isLiked) {
+    api
+      .likeCard(cardId.getId())
+      .then((res) => {
+        cardId.updateLikeStatus(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .disLikeCard(cardId.getId())
+      .then((res) => {
+        cardId.updateLikeStatus(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors